test/test.js

```diff
@@ -5,16 +5,39 @@ var wrapComponent = require('../index');
 
 describe('onclickoutside hoc', function() {
 
+  var externalToggleEnableClickOutside;
+
   var Component = React.createClass({
     getInitialState: function() {
       return {
-        clickOutsideHandled: false
+        clickOutsideHandled: false,
+        timesHandlerCalled: 0
       };
     },
 
-    handleClickOutside: function() {
+    toggleEnableClickOutside: function(on) {
+      if(on) {
+        this.props.enableOnClickOutside();
+      }
+      else {
+        this.props.disableOnClickOutside();
+      }
+    },
+
+    componentDidMount: function() {
+      externalToggleEnableClickOutside = function(on) {
+        this.toggleEnableClickOutside(on);
+      }.bind(this);
+    },
+
+    handleClickOutside: function(event) {
+      if (event === undefined) {
+          throw new Error("event cannot be undefined");
+      }
+
       this.setState({
-        clickOutsideHandled: true
+        clickOutsideHandled: true,
+        timesHandlerCalled: this.state.timesHandlerCalled + 1
```

Comment:
As this is a testing-only value it's not super important,  but it's probably best to lift this out of the state - even if we're just testing, updating the state in any way causes React to run through the `render()` pipeline, which we probably want to avoid.